### ISSUE TITLE
Proposal: Add a built-in `asArray` operation for ListOf nodes.

### ISSUE THREAD

**Issue by pdubroy** [cdglabs/ohm#62]
The built-in `ListOf` rule is helpful, but when you write a semantics for your grammar, you almost always end up writing some boilerplate actions like this:

```
NonemptyListOf(first, _, rest) {
  return [first.myAttr].concat(rest.myAttr);
},
EmptyListOf() {
  return [];
}
```

It would be nice if we had a built-in operation/attribute that would help people avoid this boilerplate. Possible names: `asArray`, `getElements`. It would just return an array of wrappers, and then you could call map to call whatever operation you want on them. E.g.: `params.asArray().map((n) => n.myAttr)`.

**Comment by mroeder** [comment]
+1

**Comment by alexwarth** [comment]
+1 from me too

This `asArray` operation should come "pre-installed" every time you create
a new semantics via `g.semantics()`.

**Comment by pdubroy** [comment]
Any strong opinions on the name? Tempted to go with asArray for now.

**Comment by alexwarth** [comment]
I'm happy with that.

**Comment by pdubroy** [comment]
This was implemented as `asIteration` in 5dfd9e4.
